File: iAsync.ObjC/lib/JFFUI/UIImage/UIImage+DrawImageInImage.m
#import "UIImage+DrawImageInImage.h"

//source: http://stackoverflow.com/questions/7313023/overlay-an-image-over-another-image-in-ios

@implementation UIImage (DrawImageInImage)

- (instancetype)drawInImage:(UIImage *)bgImage
                    atPoint:(CGPoint)point
                       size:(CGSize)size
{
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.f);
    
    [bgImage drawAsPatternInRect:CGRectMake(0.f, 0.f, size.width, size.height)];
    [self drawInRect:CGRectMake(point.x, point.y, self.size.width, self.size.height)];
    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return newImage;
}

@end

File: iAsync.Swift/lib/JUI/JUI/UIImage/UIImage+DrawImageInImage.swift
//
//  UIImage+DrawImageInImage.swift
//  JUI
//
//  Created by Vladimir Gorbenko on 19.03.15.
//  Copyright (c) 2015 EmbeddedSources. All rights reserved.
//

import Foundation

public extension UIImage {
    
    func drawInImage(bgImage: UIImage, atPoint point: CGPoint, size: CGSize) -> UIImage
    {
        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)
        
        bgImage.drawAsPatternInRect(CGRectMake(0.0, 0.0, size.width, size.height))
        drawInRect(CGRectMake(point.x, point.y, size.width, size.height))
        let newImage = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        
        return newImage
    }
}
